merlin/models/tf/core/encoder.py

```diff
@@ -145,3 +148,102 @@ def get_config(self):
             config[i] = tf.keras.utils.serialize_keras_object(layer)
 
         return config
+
+
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class TopKEncoder(EncoderBlock, BaseModel):
```

Comment:
`EncoderBlock` also inherits from the Keras-model. So what functionality is needed here from `BaseModel`?

Reply:
The functionalities needed are:
- The custom `compile` to create default loss and metrics 
-  The custom [test_step](https://github.com/NVIDIA-Merlin/models/blob/614615f8617321260a2b427be586a96eae9b9027/merlin/models/tf/models/base.py#L593) to compute loss and metrics from the prediction output layer.